### ISSUE TITLE
ChatMessageHostedContent Exception

### ISSUE THREAD

**Issue by iot-dev01** [microsoftgraph/msgraph-sdk-dotnet#1084]
**Describe the bug**
ChatMessageHostedContent Exception during send after update to 4.0.0

**To Reproduce**
Try the Sample provided by Microsoft Documentation
[https://docs.microsoft.com/en-us/graph/api/chatmessage-post?view=graph-rest-1.0&tabs=csharp#example-5-send-inline-images-along-with-the-message](https://docs.microsoft.com/en-us/graph/api/chatmessage-post?view=graph-rest-1.0&tabs=csharp#example-5-send-inline-images-along-with-the-message)

You will see the following error:
```
Code: BadRequest 
Message: The request is malformed or incorrect: expected a string for ODataType value.
```
If we add the OdataType to the ChatMessageHostedContent object - we get the error:
```
Code: BadRequest
Message: Invalid request body was sent.
```

**Expected behavior**
The image should be send to teams as before the update.

**Desktop (please complete the following information):**
Windows 10
 
[AB#10461](https://microsoftgraph.visualstudio.com/0985d294-5762-4bc2-a565-161ef349ca3e/_workitems/edit/10461)

**Comment by MIchaelMainer** [comment]
@iot-dev01 

The workaround for now will be to set the type like this:

```cs
new ChatMessageHostedContent
{
	ContentBytes = Encoding.ASCII.GetBytes("iVBORwggg=="),
	ContentType = "image/png",
	AdditionalData = new Dictionary<string, object>()
	{
		{"@microsoft.graph.temporaryId", "1"}
	},
	ODataType = "microsoft.graph.chatMessageHostedContent"
}
```

@andrueastman 

The generated code is emitting two errors. First, it is emitting a null odata.type value. No odata.type should be emitted. Second, it is emitting the unset id property from base entity. Here is what v4 is emitting:

```cs
{
  "body": {
    "content": "<div><div>\n<div><span><img height=\"297\" src=\"../hostedContents/1/$value\" width=\"297\" style=\"vertical-align:bottom; width:297px; height:297px\"></span>\n\n</div>\n\n\n</div>\n</div>",
    "contentType": "html"
  },
  "hostedContents": [
    {
      "contentBytes": "aVZCT1J3Mya0pnZ2c9PQ==",
      "contentType": "image/png",
      "id": null,
      "@odata.type": null,
      "@microsoft.graph.temporaryId": "1"
    }
  ]
}
```

Here is what v3 is emitting:

```cs
{
  "body": {
    "content": "<div><div>\n<div><span><img height=\"297\" src=\"../hostedContents/1/$value\" width=\"297\" style=\"vertical-align:bottom; width:297px; height:297px\"></span>\n\n</div>\n\n\n</div>\n</div>",
    "contentType": "html",
    "@odata.type": "microsoft.graph.itemBody"
  },
  "hostedContents": [
    {
      "contentBytes": "aVZCTa0pnZ2c9PQ==",
      "contentType": "image/png",
      "@odata.type": "microsoft.graph.chatMessageHostedContent",
      "@microsoft.graph.temporaryId": "1"
    }
  ],
  "@odata.type": "microsoft.graph.chatMessage"
}

**Comment by andrueastman** [comment]
This is resolved via [4.1.0 release](https://www.nuget.org/packages/Microsoft.Graph/4.1.0)
